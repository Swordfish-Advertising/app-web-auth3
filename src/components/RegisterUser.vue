<template>
  <v-form ref="form" v-model="validForm">
    <h1>Register a new user</h1>

    <v-text-field
      label="E-mail"
      :rules="[rules.required, rules.email]"
    ></v-text-field>

    <v-text-field
      label="Username"
      :rules="[rules.required]"
    ></v-text-field>
    
    <Password :confirmation="true"></Password>

    <v-autocomplete
      label="Hosting"
      :items="hosts"
      placeholder="Choose where to store your data..."
      :rules="[rules.required]"
    ></v-autocomplete>
    
    <div>
      By registering you agree with our
      <a target="_blank" href="https://pryv.com/terms-of-use/">terms of use</a>.
    </div>
    
    <v-btn
      @click="submit"
      :disabled="!validForm"
    >Submit</v-btn>

    <v-btn
      @click="clear"
    >Clear</v-btn>

    <NavigationButton
      :title="'Already a Pryv user ? Sign in.'"
      :page="'auth'"
    ></NavigationButton>

  </v-form>
</template>

<script>
  import Password from './Password.vue'
  import NavigationButton from './NavigationButton.vue'

  export default {
    components: {
      Password,
      NavigationButton
    },
    data: () => ({
      hosts: [ 'EU', 'US', 'ASIA' ],
      rules: {
        required: value => !!value || 'This field is required.',
        email: value => /.+@.+/.test(value) || 'E-mail must be valid'
      },
      validForm: false
    }),
    methods: {
      submit () {
        if (this.$refs.form.validate()) {
          alert('Submit form...')
        }
      },
      clear () {
        this.$refs.form.reset()
      }
    }
  }
</script>