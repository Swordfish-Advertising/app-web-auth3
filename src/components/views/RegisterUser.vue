<template>
  <v-form id="registerForm" ref="form" v-model="validForm">

    <v-alert
      :value="err"
      type="error"
      transition="scale-transition"
    >{{err}}</v-alert>

    <h1>Register a new user</h1>

    <v-text-field
      label="E-mail"
      id="email"
      v-model="email"
      :rules="[rules.required, rules.email]"
    ></v-text-field>

    <v-text-field
      label="Username"
      id="username"
      v-model="username"
      :rules="[rules.required]"
    ></v-text-field>
    
    <Password v-model="password" :confirmation="true"></Password>

    <v-autocomplete
      label="Hosting"
      id="hosting"
      v-model="hosting"
      :items="hosts"
      placeholder="Choose hosting..."
      :rules="[rules.required]"
    ></v-autocomplete>

    <div>
      By registering you agree with our
      <a target="_blank" href="https://pryv.com/terms-of-use/">terms of use</a>.
    </div>
    
    <v-btn
      id="submitButton"
      @click="submit"
      :disabled="!validForm"
    >Submit</v-btn>

    <v-btn
      id="clearButton"
      @click="clear"
    >Clear</v-btn>

    <router-link :to="{ name: 'Authorization' }">Already a Pryv user ? Sign in.</router-link>

  </v-form>
</template>

<script>
  import Password from './bits/Password.vue';
  import Pryv from '../models/Pryv.js';
  import context from '../../context.js';

  export default {
    components: {
      Password,
    },
    data: () => ({
      username: '',
      password: '',
      email: '',
      hosting: '',
      hosts: [],
      err: '',
      rules: {
        required: value => !!value || 'This field is required.',
        email: value => /.+@.+/.test(value) || 'E-mail must be valid.'
      },
      validForm: false
    }),
    async created() {
      context.pryv.setErrorHandler(err => {
        return this.err = JSON.stringify(err);
      });
      this.hosts = await context.pryv.getAvailableHostings();
    },
    methods: {
      async submit () {
        if (this.$refs.form.validate()) {
          await context.pryv.createUser(
            this.username,
            this.password,
            this.email,
            this.hosting
          );
          this.$router.push('auth');
        }
      },
      clear () {
        this.$refs.form.reset()
      }
    }
  }
</script>