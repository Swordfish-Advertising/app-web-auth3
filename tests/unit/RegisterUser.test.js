import { shallowMount } from '@vue/test-utils'
import RegisterUser from '../../src/components/RegisterUser'

describe ('RegisterUser.test.js', () => {

  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(RegisterUser)
  })

  it('renders correctly (snapshots matching)', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it('deactivates submit button when form is invalid', () => {
    wrapper.find({name: 'submit'}).element.disabled.toBe(true)
  })

  it('activates submit button when form is valid', () => {

  })

  it('validates email correctly', () => {

  })

  it('validates username correctly', () => {

  })

  it('validates password correctly', () => {

  })

  it('validates password confirmation correctly', () => {

  })

  it('validates hosting correctly', () => {

  })

  it('contains a valid link to terms of use', () => {

  })

  it('clears the form when clicking on clear button', () => {

  })

  it('redirects to auth view when clicking on sign in button', () => {

  })

})