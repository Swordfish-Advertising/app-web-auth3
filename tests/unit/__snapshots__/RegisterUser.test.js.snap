// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterUser.test.js renders correctly (snapshots matching) 1`] = `
<v-form>
  <h1>
    Register a new user
  </h1>
   
  <v-text-field
    label="E-mail"
    name="email"
    rules="function required(value) {
          return !!value || 'This field is required.';
        },function email(value) {
          return (/.+@.+/.test(value) || 'E-mail must be valid'
          );
        }"
  />
   
  <v-text-field
    label="Username"
    name="username"
    rules="function required(value) {
          return !!value || 'This field is required.';
        }"
  />
   
  <password-stub
    confirmation="true"
  />
   
  <v-autocomplete
    items="EU,US,ASIA"
    label="Hosting"
    name="hosting"
    placeholder="Choose where to store your data..."
    rules="function required(value) {
          return !!value || 'This field is required.';
        }"
  />
   
  <div>
    
    By registering you agree with our
    
    <a
      href="https://pryv.com/terms-of-use/"
      target="_blank"
    >
      terms of use
    </a>
    .
  
  </div>
   
  <v-btn
    disabled="true"
    name="submit"
  >
    Submit
  </v-btn>
   
  <v-btn
    name="clear"
  >
    Clear
  </v-btn>
   
  <navigationbutton-stub
    page="auth"
    title="Already a Pryv user ? Sign in."
  />
</v-form>
`;
